xrmocap/core/estimation/mview_mperson_end2end_estimator.py

```diff
@@ -0,0 +1,172 @@
+                 verbose: bool = True,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Api for estimating smpl parameters in a multi-view, single-person
+        scene.
+
+        Args:
+            work_dir (str):
+                Path to the folder for running the api. No file in work_dir
+                will be modified
+                or added by MultiViewSinglePersonSMPLEstimator.
+            smplify (Union[dict, SMPLify]):
+                A SMPLify instance or its config.
+            kps3d_model (Union[dict, torch.nn.module]):
+                An end-to-end mview mperson keypoints3d predicting model.
+            kps3d_optimizers (Union[
+                    List[Union[BaseOptimizer, dict]], None], optional):
+                A list of keypoints3d optimizers or their configs. If given,
+                keypoints3d will be
+                optimized by the cascaded final optimizers before estimation.
+                Defaults to None.
+            verbose (bool, optional):
+                Whether to print(logger.info) information during estimating.
+                Defaults to True.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+        super().__init__(
+            work_dir=work_dir,
+            smplify=smplify,
+            kps3d_optimizers=kps3d_optimizers,
+            verbose=verbose,
+            logger=logger)
+
+        self.load_batch_size = load_batch_size
+        if isinstance(kps3d_model, dict):
+            self.kps3d_model = build_architecture(kps3d_model)
+        else:
+            self.kps3d_model = kps3d_model
+
+        if isinstance(smplify, dict):
+            smplify['logger'] = logger
+            self.smplify = build_registrant(smplify)
+        else:
+            self.smplify = smplify
+        if kps3d_optimizers is None:
+            self.kps3d_optimizers = None
+        else:
+            self.kps3d_optimizers = []
+            for kps3d_optim in kps3d_optimizers:
+                if isinstance(kps3d_optim, dict):
+                    kps3d_optim['logger'] = logger
+                    kps3d_optim = build_keypoints3d_optimizer(kps3d_optim)
+                self.kps3d_optimizers.append(kps3d_optim)
+
+    @overload
+    def run(
+        self, img_arr: np.ndarray, cam_param: List[FisheyeCameraParameter]
+    ) -> Tuple[List[Keypoints], Keypoints, SMPLData]:
+        ...
+
+    @overload
+    def run(
+        self, img_paths: List[List[str]],
+        cam_param: List[FisheyeCameraParameter]
+    ) -> Tuple[List[Keypoints], Keypoints, SMPLData]:
+        ...
+
+    @overload
+    def run(
+        self, video_path: List[str], cam_param: List[FisheyeCameraParameter]
+    ) -> Tuple[List[Keypoints], Keypoints, SMPLData]:
+        ...
+
+    def run(
+        self,
+        cam_param: List[FisheyeCameraParameter],
```

Comment:
For mvp model inference, we need to convert `FisheyeCameraParameter` into a `dict` of meta. It was done in `dataset` before and some extra info such as affine transformation is also needed. Shall we do this somewhere else and pass the `dict` directly instead of `FisheyeCameraParameter`?

Reply:
The `run()` method of an estimator is designed for users who may not care about how our algorithm works. For algorithm, we use our special arguments, which is the meta-dict. But for a user interface, it should be easy to prepare, and share the same definition with other estimator. I suggest that we do conversion inside run.

Reply:
I see. If we have to do all the conversion and image transformation inside `run()`, we can pass the `dataset config` dict as a part of `kps3d_optimizers`, and prepare the data like what we did in model evaluation.